### PR TITLE
Sort Product

### PR DESCRIPTION
**Pull request by tavisca-acharpe**

### CODE REVIEW

File: src/OnlineRetailPortal.Core/Models/Product.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public static async Task<ProductsWithPageInitiation> GetProductsAsync(GetProduct
             var getProductsEntity = serviceRequest.ToEntity();
             var getProductsResponse = await productStore.GetProductsAsync(getProductsEntity);
             var coreProductResponse = getProductsResponse.ToModel();
-            coreProductResponse.PagingInfo = new PagingInfo() { PageNumber = 1, PageSize = 10, TotalPages = 100 };
+           // coreProductResponse.PagingInfo = new PagingInfo() { PageNumber = 1, PageSize = 10, TotalPages = 100 };
```
Comment (tavisca-swapnil) [inline src/OnlineRetailPortal.Core/Models/Product.cs]: Please remove comment

File: src/OnlineRetailPortal.Core/Translators/GetProductsStoreRequestTranslator.cs [hunk 2]
```diff
@@ -6,14 +6,15 @@ public static class GetProductServiceRequestTranslator
     {
         public static GetProductsEntity ToEntity(this GetProductsServiceRequest request)
         {
-            if (request.PagingInfo == null)
+            if (request.PagingInfo == null && request.ProductSort == null)
```
Comment (tavisca-swapnil) [inline src/OnlineRetailPortal.Core/Translators/GetProductsStoreRequestTranslator.cs]: Please check logic here.

**Review by tavisca-swapnil — APPROVED** [review]

File: src/OnlineRetailPortal.Core/Translators/GetProductsStoreRequestTranslator.cs (hunk 2 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Core/Translators/GetProductsStoreRequestTranslator.cs]: done

File: src/OnlineRetailPortal.Core/Models/Product.cs (hunk 1 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Core/Models/Product.cs]: done

File: src/OnlineRetailPortal.Mock/Filter.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using OnlineRetailPortal.Contracts;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OnlineRetailPortal.Mock
+{
+    public static class Filter
+    {
+        public static List<Product> ApplyFilter(this List<Product> productList, List<Contracts.Filter> filters)
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Mock/Filter.cs]: rename it to `Apply`

File: src/OnlineRetailPortal.Mock/Sort.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public static class Sort
+    {
+        public static List<Product> ApplySort(this List<Product> productList , Contracts.Sort sort)
+        {
+            if (sort == null)
+                return productList;
+            return sort.Order == "Asc" ? productList.SortAsc(sort.Type) : productList.SortDesc(sort.Type);
+
+        }
+        public static List<Product> SortAsc(this List<Product> products, string sortBy)
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Mock/Sort.cs:18]: club both the methods as single sort function which will sort the products based on inputs.
SortAsc and SortDesc as public methods are not really needed.
You can have one method which sorts the data and then delegates the sorting to these private methods

File: src/OnlineRetailPortal.Web/Models/GetProductsRequest.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+
+namespace OnlineRetailPortal.Web
+{
+    [DataContract]
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Web/Models/GetProductsRequest.cs]: why data contract attribute ?

File: src/OnlineRetailPortal.Web/Models/Money.cs [hunk 6]
```diff
@@ -4,10 +4,5 @@ public class Money
     {
         public double Amount { get; set; } 
         public string Currency { get; set; }
-        //public Money(double amount, string currency)
-        //{
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Web/Models/Money.cs:8]: I want to understand why are we sending the money object in response ?
Please loop in people which have worked on this

File: src/OnlineRetailPortal.Web/Translators/GetProductsServiceRequestTranslator.cs [hunk 7]
```diff
@@ -10,15 +10,36 @@ public static class GetProductsServiceRequestTranslator
...
                 {
                     PageNumber = pageNo == 0 ? _pageNo : pageNo,
                     PageSize = pageSize == 0 ? _pageSize : pageSize
+                },
+                ProductSort = new Contracts.Sort()
+                {
+                    Type = String.IsNullOrEmpty(request.ProductSort.Type) ? _sortType : request.ProductSort.Type,
+                    Order = String.IsNullOrEmpty(request.ProductSort.Order) ? _sortOrder : request.ProductSort.Order
                 }
+                //Filter is not implemented
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Web/Translators/GetProductsServiceRequestTranslator.cs]: remove commented code

File: src/OnlineRetailPortal.Mock/Filter.cs (hunk 3 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Mock/Filter.cs]: done

File: src/OnlineRetailPortal.Web/Models/GetProductsRequest.cs (hunk 5 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Web/Models/GetProductsRequest.cs]: Removed

File: src/OnlineRetailPortal.Web/Translators/GetProductsServiceRequestTranslator.cs (hunk 7 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Web/Translators/GetProductsServiceRequestTranslator.cs]: done

**Review by Tavisca-smohite — APPROVED** [review]

**Pull request merged**